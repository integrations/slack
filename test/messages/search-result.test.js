const SearchResult = require('../../lib/messages/search-result');
const fixtures = require('../fixtures');

const resource = { owner: 'kubernetes', repo: 'kubernetes' };

describe('Search results rendering', () => {
  test('renders results', () => {
    const searchResultMessage = new SearchResult(fixtures.searchResult.success.items, resource);
    expect(searchResultMessage.toJSON()).toMatchSnapshot();
  });
  test('handles empty results', () => {
    const searchResultMessage = new SearchResult(fixtures.searchResult.empty.items, resource);
    expect(searchResultMessage.toJSON()).toMatchSnapshot();
  });
});
