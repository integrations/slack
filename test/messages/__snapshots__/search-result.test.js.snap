// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search results rendering handles empty results 1`] = `
Object {
  "text": "No search results found for \`kubernetes/kubernetes\`",
}
`;

exports[`Search results rendering renders results 1`] = `
Object {
  "attachments": Array [
    Object {
      "fallback": "[kubernetes/kubernetes] pkg/apis/batch/v1beta1/zz_generated.conversion.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/d08e68e75974eb31fd65422c969b352ed8397edc/pkg/apis/batch/v1beta1/zz_generated.conversion.go",
      "title": "zz_generated.conversion.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/d08e68e75974eb31fd65422c969b352ed8397edc/pkg/apis/batch/v1beta1/zz_generated.conversion.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] pkg/apis/batch/v2alpha1/zz_generated.conversion.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/d08e68e75974eb31fd65422c969b352ed8397edc/pkg/apis/batch/v2alpha1/zz_generated.conversion.go",
      "title": "zz_generated.conversion.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/d08e68e75974eb31fd65422c969b352ed8397edc/pkg/apis/batch/v2alpha1/zz_generated.conversion.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/kubernetes/typed/batch/v1beta1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v1beta1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v1beta1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/kubernetes/typed/batch/v2alpha1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v2alpha1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v2alpha1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] test/e2e/apps/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/test/e2e/apps/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/test/e2e/apps/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/deprecated/typed/batch/v1beta1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v1beta1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v1beta1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/deprecated/typed/batch/v2alpha1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v2alpha1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v2alpha1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/listers/batch/v1beta1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/client-go/listers/batch/v1beta1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/client-go/listers/batch/v1beta1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/listers/batch/v2alpha1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/client-go/listers/batch/v2alpha1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/client-go/listers/batch/v2alpha1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] pkg/registry/batch/cronjob/strategy.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/5ebdf3e3f211720b50dc0cb5dc727201ac2688e4/pkg/registry/batch/cronjob/strategy.go",
      "title": "strategy.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/5ebdf3e3f211720b50dc0cb5dc727201ac2688e4/pkg/registry/batch/cronjob/strategy.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/kubernetes/typed/batch/v1beta1/fake/fake_cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v1beta1/fake/fake_cronjob.go",
      "title": "fake_cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v1beta1/fake/fake_cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/kubernetes/typed/batch/v2alpha1/fake/fake_cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v2alpha1/fake/fake_cronjob.go",
      "title": "fake_cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/staging/src/k8s.io/client-go/kubernetes/typed/batch/v2alpha1/fake/fake_cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/deprecated/typed/batch/v1beta1/fake/fake_cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v1beta1/fake/fake_cronjob.go",
      "title": "fake_cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v1beta1/fake/fake_cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/deprecated/typed/batch/v2alpha1/fake/fake_cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v2alpha1/fake/fake_cronjob.go",
      "title": "fake_cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/ab97b666e214161bf5b694badab1030333c7d6cc/staging/src/k8s.io/client-go/deprecated/typed/batch/v2alpha1/fake/fake_cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] pkg/registry/batch/cronjob/strategy_test.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/52705deb62d9b597eb38b119d466a78db89e5eb1/pkg/registry/batch/cronjob/strategy_test.go",
      "title": "strategy_test.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/52705deb62d9b597eb38b119d466a78db89e5eb1/pkg/registry/batch/cronjob/strategy_test.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v1beta1/zz_generated.deepcopy.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/d8fd232ea1f8c91092fb5fabb7a0f3d557a2e8fb/staging/src/k8s.io/api/batch/v1beta1/zz_generated.deepcopy.go",
      "title": "zz_generated.deepcopy.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/d8fd232ea1f8c91092fb5fabb7a0f3d557a2e8fb/staging/src/k8s.io/api/batch/v1beta1/zz_generated.deepcopy.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v2alpha1/zz_generated.deepcopy.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/d8fd232ea1f8c91092fb5fabb7a0f3d557a2e8fb/staging/src/k8s.io/api/batch/v2alpha1/zz_generated.deepcopy.go",
      "title": "zz_generated.deepcopy.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/d8fd232ea1f8c91092fb5fabb7a0f3d557a2e8fb/staging/src/k8s.io/api/batch/v2alpha1/zz_generated.deepcopy.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v1beta1/types_swagger_doc_generated.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v1beta1/types_swagger_doc_generated.go",
      "title": "types_swagger_doc_generated.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v1beta1/types_swagger_doc_generated.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v2alpha1/types_swagger_doc_generated.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v2alpha1/types_swagger_doc_generated.go",
      "title": "types_swagger_doc_generated.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v2alpha1/types_swagger_doc_generated.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] test/e2e/apimachinery/generated_clientset.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/test/e2e/apimachinery/generated_clientset.go",
      "title": "generated_clientset.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/test/e2e/apimachinery/generated_clientset.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/informers/batch/v1beta1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/dde6e8e7465468c32642659cb708a5cc922add64/staging/src/k8s.io/client-go/informers/batch/v1beta1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/dde6e8e7465468c32642659cb708a5cc922add64/staging/src/k8s.io/client-go/informers/batch/v1beta1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/client-go/informers/batch/v2alpha1/cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/dde6e8e7465468c32642659cb708a5cc922add64/staging/src/k8s.io/client-go/informers/batch/v2alpha1/cronjob.go",
      "title": "cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/dde6e8e7465468c32642659cb708a5cc922add64/staging/src/k8s.io/client-go/informers/batch/v2alpha1/cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v1beta1/types.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v1beta1/types.go",
      "title": "types.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v1beta1/types.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v2alpha1/types.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v2alpha1/types.go",
      "title": "types.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v2alpha1/types.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v1beta1/generated.proto",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v1beta1/generated.proto",
      "title": "generated.proto",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v1beta1/generated.proto",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/api/batch/v2alpha1/generated.proto",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v2alpha1/generated.proto",
      "title": "generated.proto",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/a9f0db16614ae62563ead2018f1692407bd93d8f/staging/src/k8s.io/api/batch/v2alpha1/generated.proto",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] pkg/registry/batch/cronjob/storage/storage_test.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/890b50f98bc981275993b5fdc9b5d450364b2a42/pkg/registry/batch/cronjob/storage/storage_test.go",
      "title": "storage_test.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/890b50f98bc981275993b5fdc9b5d450364b2a42/pkg/registry/batch/cronjob/storage/storage_test.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] pkg/apis/batch/zz_generated.deepcopy.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/d8fd232ea1f8c91092fb5fabb7a0f3d557a2e8fb/pkg/apis/batch/zz_generated.deepcopy.go",
      "title": "zz_generated.deepcopy.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/d8fd232ea1f8c91092fb5fabb7a0f3d557a2e8fb/pkg/apis/batch/zz_generated.deepcopy.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/fb6f5d739b08e049b188806d6e33099f85fad2fe/staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go",
      "title": "create_cronjob.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/fb6f5d739b08e049b188806d6e33099f85fad2fe/staging/src/k8s.io/kubectl/pkg/cmd/create/create_cronjob.go",
    },
    Object {
      "fallback": "[kubernetes/kubernetes] test/integration/cronjob/cronjob_test.go",
      "footer": "kubernetes/kubernetes",
      "footer_icon": "https://github.githubassets.com/favicon.ico",
      "text": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/test/integration/cronjob/cronjob_test.go",
      "title": "cronjob_test.go",
      "title_link": "https://github.com/kubernetes/kubernetes/blob/abe6321296123aaba8e83978f7d17951ab1b64fd/test/integration/cronjob/cronjob_test.go",
    },
  ],
}
`;
